MetaMask/metamask-extension
# Adding option to set Custom Nonce to Confirm Approve Page

Fixes: #10129

Explanation:

The option to set custom nonce was not available in Confirm Approve Page when the `Customize transaction nonce` is `ON`. 

This was because `<ConfirmApproveContent/>` is passed as the `contentComponent` propertery of `<ConfirmTransactionBase/>` which prevents  rendering of `renderDetails()` (method in `Confirm-Transaction-Base.component.js`), which contains the code for Custom Nonce settings.

The new changes includes a section to display the `nextNonce` number in the approve screen after `Transaction Fee` and before the collapsing section with `Edit` button. Upon clicking the `edit`  a new modal will open enabling the user to update the nonce number.

Manual testing steps:  
  - Set `Customize transaction nonce` to 'ON'
  -  From the [test-dapp](https://metamask.github.io/test-dapp/) connect to the wallet and `Create Token`
  - Once the Token is added, Click `Approve token`.
  - In the `Approve Token window` click the `edit` button the nonce section , a new modal will appear where the user can customize the nonce value.

https://user-images.githubusercontent.com/43930900/115086079-9633c880-9ed9-11eb-9281-9162ccce3ed4.mov
